src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -172,11 +173,13 @@ def __init__(
 
     def _init_index(
         self,
-        data: ArrayLike | native_pd.Index | Series | None,
+        data: ArrayLike | native_pd.Index | Series | DataFrame | None,
```

Comment:
```suggestion
        data: ArrayLike | native_pd.Index | Series | None,
```
Can not create index from dataframe